rewrite-java/src/main/java/org/openrewrite/java/ChangeType.java

```diff
@@ -491,6 +491,7 @@ private static class ChangeClassDefinition extends JavaIsoVisitor<ExecutionConte
         private final JavaType.Class originalType;
         private final JavaType.Class targetType;
         private final MethodMatcher originalConstructor;
+        private boolean updatePath;
```

Comment:
We generally recommend to [use cursor messaging to pass state](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices#use-cursor-messaging-instead-of-execution-context-messaging) between different methods of a visitor. Mutable fields might linger with old values, whereas cursor messages follow the visitor pattern, and get reset as needed.

Reply:
Changed - polling when checking should be the correct behavior, though I'm not 100% on that